### PR TITLE
[IMP] models: show better server error message

### PR DESCRIPTION
**Pull request by jvh83**
This will show better and clearer error message in case of access rights issues

### CODE REVIEW

**Comment by fmdl** [comment]
this is done in master no ? https://github.com/odoo/odoo/pull/30144

File: addons/mail/models/mail_message.py [hunk 1]
```diff
@@ -595,8 +595,8 @@ def _generate_model_record_ids(msg_val, msg_ids):
                                 WHERE message.message_type = %%s AND (message.subtype_id IS NULL OR subtype.internal IS TRUE) AND message.id = ANY (%%s)''' % (self._table), ('comment', self.ids,))
             if self._cr.fetchall():
                 raise AccessError(
-                    _('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s)') %
-                    (self._description, operation))
+                    _('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s, Records: %s, User: %s)') %
+                    (self._description, operation, self.ids ,self_uid))
```
Comment (nim-odoo) [inline addons/mail/models/mail_message.py]: `self._uid`

**Comment by fmdl** [comment]
Warning, the number of `%s` have been changed, this will create errors with translation, no ?

cc @odony

File: odoo/models.py [hunk 2]
```diff
@@ -2859,8 +2858,8 @@ def _check_record_rules_result_count(self, result_ids, operation):
                 if self._uid == SUPERUSER_ID:
                     return
                 _logger.info('Access Denied by record rules for operation: %s on record ids: %r, uid: %s, model: %s', operation, forbidden_ids, self._uid, self._name)
-                raise AccessError(_('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s)') % \
-                                    (self._description, operation))
+                raise AccessError(_('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s, Records: %r, User: %s)') %
```
Comment (nim-odoo) [inline odoo/models.py]: Why using `%r` in this specific case?

File: addons/website_mail/i18n/website_mail.pot [hunk 3]
```diff
@@ -159,7 +159,7 @@ msgstr ""
 #, python-format
 msgid "The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n"
 "\n"
-"(Document type: %s, Operation: %s)"
+"(Document type: %s, Operation: %s, Records: %r, User: %s)"
```
Comment (nim-odoo) [inline addons/website_mail/i18n/website_mail.pot]: Wrong copy-paste...

File: addons/mail/i18n/mail.pot [hunk 4]
```diff
@@ -4632,7 +4632,7 @@ msgstr ""
 #, python-format
 msgid "The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n"
 "\n"
-"(Document type: %s, Operation: %s)"
+"(Document type: %s, Operation: %s, Records: %r, User: %s)"
```
Comment (nim-odoo) [inline addons/mail/i18n/mail.pot]: Wrong copy-paste

**Comment by nim-odoo** [comment]
> Warning, the number of `%s` have been changed, this will create errors with translation, no ?
> 
> cc @odony

This will break the translations (coucou @mart-e ).

**Comment by fmdl** [comment]
@mart-e in master there allready this imp : #30144, it could be nice to use this for accesserror in mail, ...

File: addons/mail/models/mail_message.py (hunk 1 above)
Comment (odony) [reply addons/mail/models/mail_message.py]: and lint!

**Comment by nim-odoo** [comment]
> @mart-e in master there allready this imp : #30144, it could be nice to use this for accesserror in mail, ...

We are aware of this, we already had internal discussions about it.

**Comment by odony** [comment]
Unrelated, but could you configure both your github account name and your local `git config user.name`?

**Pull request closed without merging**